app/src/main/java/com/deneyehayir/deneysiz/scene/MainContent.kt

```diff
@@ -3,50 +3,26 @@ package com.deneyehayir.deneysiz.scene
 import androidx.compose.foundation.layout.padding
 import androidx.compose.material.Scaffold
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.SideEffect
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
-import androidx.navigation.compose.currentBackStackEntryAsState
-import androidx.navigation.compose.rememberNavController
 import com.deneyehayir.deneysiz.ui.theme.DeneysizTheme
-import com.deneyehayir.deneysiz.ui.theme.Orange
-import com.deneyehayir.deneysiz.ui.theme.White0
-import com.google.accompanist.systemuicontroller.rememberSystemUiController
 
 @Composable
-fun MainContent() {
+fun MainContent(
+    appState: MainAppState = rememberMainAppState()
```

Comment:
The improvements to managing app state and navigation look great, but I found some weird flickering in Category Screens when I use the app. I think these side effects are due to improvements in navigation logic. Maybe we should consider adding the improvements in a separate branch.